app/server/src/server/framework-preset-server.ts

```diff
@@ -0,0 +1,24 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { Configuration } from 'webpack';
+import path from 'path';
+
+export function webpack(config: Configuration) {
+  return {
+    ...config,
+    module: {
+      ...config.module,
+      rules: [
+        ...config.module.rules,
+        {
+          type: 'javascript/auto',
+          test: /\.stories\.json$/,
+          use: [
+            {
+              loader: path.resolve(__dirname, './loader.js'),
+            },
+          ],
+        },
+      ],
+    },
+  };
```

Comment:
So is the idea that you add the `x.stories.json` files to your storybook directory? What about the idea of fetching the json on startup?

Reply:
Discussed this and the answer is it is better to let the external process (i.e. Rails) watch the relevant files and regenerate the JSON file, triggering webpack to rebuild when the stories change.